packages/protocol/contracts/signal/ISignalService.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+//
+// ╭━━━━╮╱╱╭╮╱╱╱╱╱╭╮╱╱╱╱╱╭╮
+// ┃╭╮╭╮┃╱╱┃┃╱╱╱╱╱┃┃╱╱╱╱╱┃┃
+// ╰╯┃┃┣┻━┳┫┃╭┳━━╮┃┃╱╱╭━━┫╰━┳━━╮
+// ╱╱┃┃┃╭╮┣┫╰╯┫╭╮┃┃┃╱╭┫╭╮┃╭╮┃━━┫
+// ╱╱┃┃┃╭╮┃┃╭╮┫╰╯┃┃╰━╯┃╭╮┃╰╯┣━━┃
+// ╱╱╰╯╰╯╰┻┻╯╰┻━━╯╰━━━┻╯╰┻━━┻━━╯
+pragma solidity ^0.8.9;
+
+interface ISignalService {
+    /**
+     * Send a signal by storing the key with a value of 1.
+     *
+     * @param user The user address sending the signal.
+     * @param signal The signal to send.
+     */
+    function sendSignal(address user, bytes32 signal) external;
+
+    /**
+     * Check if a signal has been sent (key stored with a value of 1).
+     *
+     * @param user The sender address of the signal.
```

Comment:
looks like the `@param` description of `app` is missed here? other part of the changes LGTM.